### PR TITLE
ddl: fix the set's  default value where `create table`

### LINKED ISSUE #12160: SET type has inconsistent behavior compared with MySQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
create table myset2(i int, s set('4','5', '6') default 4);
insert into myset2(i) values(1);
```
2. What did you expect to see?
```
mysql>  select * from myset2;
+------+------+
| i    | s    |
+------+------+
|    1 | 6    |
+------+------+
1 row in set (0.01 sec)
```

3. What did you see instead?
```
mysql>  select * from myset2;
+------+------+
| i    | s    |
+------+------+
|    1 | 4    |
+------+------+
1 row in set (0.01 sec)
```
MySQL's show create table is the following 
```
mysql> show create table myset2;
+--------+-----------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                                |
+--------+-----------------------------------------------------------------------------------------------------------------------------+
| myset2 | CREATE TABLE `myset2` (
  `i` int(11) DEFAULT NULL,
  `s` set('4','5','6') DEFAULT '6'
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+-----------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

while tidb's is different

```
mysql> show create table myset2;
+--------+-----------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                                |
+--------+-----------------------------------------------------------------------------------------------------------------------------+
| myset2 | CREATE TABLE `myset2` (
  `i` int(11) DEFAULT NULL,
  `s` set('4','5','6') DEFAULT '4'
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+-----------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 

- Fix #12160. Fix the issue when the set type default value is int type. 

- Fix an issue where the default value does not exist in the set.

### What is changed and how it works?

- Call `ParseSetValue` when the set type default value is int type. Mysql implementation also has problems, as follows:

```
mysql> create table t_set (a int, b set('1', '5') default 1);
Query OK, 0 rows affected (0.02 sec)
mysql> create table t (a int, b set('1', '5') default 2);
ERROR 1067 (42000): Invalid default value for 'b'
mysql> create table t (a int, b set('4', '5') default 1);
```
So code implementation reference [MySQL set type](https://dev.mysql.com/doc/refman/5.7/en/set.html).

- Check if the default value exists in the set.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -698,6 +698,57 @@ func getDefaultValue(ctx sessionctx.Context, colName string, c *ast.ColumnOption
 	return v.ToString()
 }
 
+func setSetDefaultValue(v types.Datum, col *table.Column) (string, error) {
```
Comment (zhexuany) [inline ddl/ddl_api.go]: Should we attache the desc of SET type for a reference?

File: go.mod [hunk 2]
```diff
@@ -70,6 +70,7 @@ require (
 	golang.org/x/text v0.3.2
 	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4 // indirect
 	golang.org/x/tools v0.0.0-20190911022129-16c5e0f7d110
+	google.golang.org/appengine v1.4.0 // indirect
```
Comment (zhexuany) [inline go.mod]: Why do you have this change? It should be irrelevant regarding of this patch.

**Comment by zimulala** [comment]
PTAL @zhexuany

**Comment by zimulala** [comment]
PTAL  @crazycs520 @bb7133

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -698,6 +698,58 @@ func getDefaultValue(ctx sessionctx.Context, colName string, c *ast.ColumnOption
...
+	str, err := v.ToString()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if str == "" {
+		return str, nil
+	}
+
+	valMap := make(map[string]struct{}, len(col.Elems))
+	dVals := strings.SplitN(strings.ToLower(str), ",", -1)
```
Comment (lonng) [inline ddl/ddl_api.go]: Prefer: 
```suggestion
	dVals := strings.Split(strings.ToLower(str), ",")
```

**Comment by bb7133** [comment]
LGTM, please address @lonng 's comment

**Pull request merged**